sound/soc/sof/intel/hda-stream.c

```diff
@@ -570,14 +587,18 @@ static bool hda_dsp_stream_check(struct hdac_bus *bus, u32 status)
 			snd_hdac_stream_writeb(s, SD_STS, sd_status);
 
 			active = true;
-			if (!s->substream ||
+			if ((!s->substream || s->cstream) ||
```

Comment:
did you mean if (!(s->substream || s->stream))?

Reply:
hdac_stream contains substream field of struct snd_pcm_substream type. I've added compr equivalent: cstream of struct snd_compr_stream type in one of the early patches. This modifications ensures "one of the runtimes" is valid before moving on.
My patches add so-called compr "support" for HDA. With them, substream is no longer a requirement as cstream can take its role.

Reply:
the code does not test that 'one of the runtimes is valid'.

Reply:
I see now what you meant. My eyes are failing me as I'm constantly swapping between cAVS and SOF solution. Thanks for spotting this out.